src/listeners/ticket/create-modal-submit-validator.ts

```diff
@@ -0,0 +1,183 @@
+import { ApplyOptions } from '@sapphire/decorators';
+import { Events, Listener } from '@sapphire/framework';
+import {
+    ActionRowBuilder,
+    ButtonBuilder,
+    ButtonStyle,
+    ChannelType,
+    EmbedBuilder,
+    GuildMember,
+    PermissionFlagsBits,
+    type Interaction,
+    type ModalSubmitInteraction,
+} from 'discord.js';
+import { Color } from '../../lib/embeds';
+import { TButtonId, InputId, ModalId } from '../../lib/ticket';
+import { messageLogger } from '../..';
+import { settingsManager } from '../..';
+import { userUrl } from '../../lib/user';
+import { ticketLogs } from '../..';
+import type { TicketLog } from '../../lib/ticket';
+
+@ApplyOptions<Listener.Options>({ event: Events.InteractionCreate })
+export class InteractionCreateListener extends Listener<
+    typeof Events.InteractionCreate
+> {
+
+    public override async run(interaction: Interaction) {
+        if (
+            !interaction.isModalSubmit() ||
+            interaction.customId !== ModalId.Ticket ||
+            !interaction.inGuild()
+        ) {
+            return;
+        }
+
+        await interaction.deferReply({ ephemeral: true });
+
+        const title = interaction.fields.getTextInputValue(InputId.Title).trim();
+        if (!title) {
+            return this.sendValidationFailure(
+                interaction,
+                'Title must contain at least 1 non-whitespace character'
+            );
+        }
+
+        const exp = interaction.fields.getTextInputValue(InputId.Explanation).trim();
+        if (!exp) {
+            return this.sendValidationFailure(
+                interaction,
+                'Description must contain at least 1 non-whitespace character'
+            );
+        }
+
+        const guildSettings = await settingsManager.get(interaction.guildId);
+        const guild = await interaction.client.guilds.fetch(interaction.guildId);
+        const member = await guild.members.fetch(interaction.member.user.id);
```

Comment:
Prefer moving these fetches further down, closer to where the result is used. There are a number of code paths between the fetching of the guild/member and their first uses that return, meaning the result of the fetches might end up being completely unused, i.e. we end up doing expensive work that we didn't have to.